packages/ui/src/Button.tsx

```diff
@@ -45,28 +45,47 @@ export type ButtonAccessibleIconRightProps =
       iconRightClassName?: never
     }
 
-export type ButtonDisabledProps =
-  | { disabledTooltip: string; disabled: boolean; disabledTooltipVisible?: boolean; disabledTooltipPlacement?: TooltipProps['placement'] }
-  | {
-      disabled?: never
-      disabledTooltip?: never
-      disabledTooltipVisible?: never
-      disabledTooltipPlacement?: never
-    }
+type ButtonNotDisabledProps = {
+  disabled?: never
+  disabledTooltip?: never
+  disabledTooltipVisible?: never
+  disabledTooltipPlacement?: never
+}
+
+export type ButtonDisabledProps = {
+  disabledTooltip: string
+  disabled: boolean
+  disabledTooltipVisible?: boolean
+  disabledTooltipPlacement?: TooltipProps['placement']
+}
 
 // Common props for all button "kinds"
 type ButtonCommonProps = {
   kind?: ButtonKind
   children: string
   capitalize?: boolean
   bodyClassName?: string
-}
+} & Pick<HTMLAttributes<HTMLAnchorElement | HTMLButtonElement>, 'className' | 'onClick'>
+
+type ButtonTypeProps =
+  | ({
+      component: 'a'
+      href: string
+    } & ButtonNotDisabledProps)
+  | ((
+      | {
+          component: 'button'
+          href?: never
+        }
+      | {
+          component?: undefined
+          href?: never
+        }
+    ) &
+      (ButtonDisabledProps | ButtonNotDisabledProps) &
+      Pick<ButtonHTMLAttributes<HTMLButtonElement>, 'autoFocus' | 'type'>)
 
-export type ButtonProps = ButtonCommonProps &
-  ButtonAccessibleIconLeftProps &
-  ButtonAccessibleIconRightProps &
-  ButtonDisabledProps &
-  Pick<ButtonHTMLAttributes<HTMLButtonElement>, 'onClick' | 'className' | 'autoFocus' | 'type'>
+export type ButtonProps = ButtonCommonProps & ButtonAccessibleIconLeftProps & ButtonAccessibleIconRightProps & ButtonTypeProps
```

Comment:
So we're allowing icons when we use `<a>` - how does it look? Do we want to allow it?

Reply:
Nope we don't, will fix, nice catch, I'll fix.

Reply:
@danad02 my bad, we actually do allow this :) Well visually it's still a button and you can use an icon.

Reply:
yes, you're right :smile: